router/router.go

```diff
@@ -27,6 +28,17 @@ import (
 func Create(db *database.GormDatabase, vInfo *model.VersionInfo, conf *config.Configuration) (*gin.Engine, func()) {
 	g := gin.New()
 
+	if conf.Server.TrustedProxies != nil {
+		g.SetTrustedProxies(conf.Server.TrustedProxies)
```

Comment:
Hmm the handling of this in gin really sucks. I'm inclined to not use any of gin for this and do custom client ip parsing. This would also fix the security bug described in https://github.com/gotify/server/issues/554#issuecomment-1560059081

What do you think about this?

```go
gin.LoggerWithFormatter(useXRealIP(conf.Server.TrustXRealIP, logFormatter))
```
```go
func useXRealIP(trustedProxy bool, inner gin.LogFormatter) gin.LogFormatter {
	return func(params gin.LogFormatterParams) string {
		params.ClientIP = getClientIp(trustedProxy, params.Request)
		return inner(params)
	}
}

func getClientIp(trustedProxy bool, req *http.Request) string {
	if trustedProxy {
		realIpParts := strings.SplitN(req.Header.Get("x-real-ip"), ",", 2)
		if ip := strings.TrimSpace(realIpParts[0]); ip != "" {
			return ip
		}
	}

	addr := req.RemoteAddr
	if addr == "@" {
		return "socket"
	} else if host, _, err := net.SplitHostPort(addr); err == nil {
		return host
	}
	return addr
}
```
x-real-ip is used because it's already defined in the gotify reverse proxy docs: https://gotify.net/docs/nginx

Reply:
Interesting the only issue is that we don't actually test if the upstream proxy is trusted, so if a user misconfigures and exposes the gotify port they could get spoofed.

Reply:
The same is possible when a user configures 0.0.0.0/0 as trusted proxy and the user still has to enable a trustXRealIP setting. Either way, I don't think this is a big problem, because the ip is only logged and not used for anything else in gotify.

Reply:
I agree users are users 😅 I'm happy with above since it won't functionally change Gin and just for logging